lusidtools/cocoon/utilities.py

```diff
@@ -296,33 +374,45 @@ def get_swagger_dict(api_url) -> dict:
         )
 
 
+def generate_required_attributes_list():
+    pass
+
+
+@checkargs
 def verify_all_required_attributes_mapped(
-    mapping, model_object, exempt_attributes=[]
+    mapping: dict, model_object_name: str, exempt_attributes: list = None
 ) -> None:
     """
     Verifies that all required attributes are included in the mapping, passes silently if they are and raises an exception
     otherwise
 
     :param dict mapping: The required mapping
-    :param lusid.model model_object: The LUSID model that the mapping applies to
+    :param str model_object_name: The name of the lusid.models object that the mapping is for
     :param list[str] exempt_attributes: The attributes that are exempt from needing to be in the required mapping
 
     :return: None
     """
+
+    # Check that the provided model name actually exists
+    model_object = getattr(lusid.models, model_object_name, None)
+
+    if model_object is None:
+        raise TypeError("The provided model_object is not a lusid.model object")
+
+    # Convert a None to an empty list
+    exempt_attributes = Validator(exempt_attributes, "exempt_attributes").set_default_value_if_none([]).value
+
+    # Gets the required attributes for this model
     required_attributes = get_required_attributes_model_recursive(
         model_object=model_object
     )
 
+    # Removes the exempt attributes
     for attribute in required_attributes:
+        # Removes all nested attributes for example if "identifiers" is exempt "identifiers.value" will be removed
         if attribute.split(".")[0] in exempt_attributes:
```

Comment:
What do you mean? I've update this so that that this function takes a key_separator argument which is passed into the function that generates these keys so that they will always match.